packages/styled/types/tsconfig.json

```diff
@@ -2,6 +2,7 @@
   "compilerOptions": {
     "baseUrl": "../",
     "forceConsistentCasingInFileNames": true,
+    "isolatedModules": true,
```

Comment:
do you see any downsides of using this in here? I can't think of any, cc @JakeGinnivan

Reply:
`isolatedModules` will cause the compilation to emit errors if the code is not able to be transpiled in single-file mode (ie. babel or ts.transpile). So it will raise errors for doing type re-exports or using const enums. AFAIK all other errors will also be emitted so I think it is safe to keep it turned on and it may even be worth adding it to the other configuration files.

Reply:
Some info about the impact here: https://github.com/microsoft/TypeScript/issues/16351

Reply:
> it may even be worth adding it to the other configuration files.

Yeah, I was thinking about this as well.

Reply:
Makes sense to me 👍

Reply:
I can add that to the PR if you want when I get home this evening.

Reply:
Merged this one in as we can add this flag in other configs in separate PR. Thanks for the fix!